cli/unipipe/commands/terraform.ts

```diff
@@ -0,0 +1,196 @@
+
+  const instances = await repo.mapInstances(async (instance) => {
+    return await instance;
+  });
+
+  return await Promise.all(
+    instances.map((instance) => mapBindings(instance, repo)),
+  );
+}
+
+async function mapBindings(
+  instance: ServiceInstance,
+  repo: Repository,
+): Promise<string[]> {
+  return await Promise.all(
+    instance.bindings.map(async(binding) => {
+      try {
+        return await processBinding(instance, binding, repo);
+      } catch (e) {
+        const bindingIdentifier = instance.instance.serviceInstanceId + "/" +
+          binding.binding.bindingId;
+
+        console.error(
+          `processing binding ${bindingIdentifier} failed with error: ${e}`,
+        );
+
+        const status: OsbServiceInstanceStatus = {
+          status: "failed",
+          description: "Processing the binding failed!",
+        };
+
+        repo.updateInstanceStatus(
+          instance.instance.serviceInstanceId,
+          status,
+        );
+
+        repo.updateBindingStatus(
+          instance.instance.serviceInstanceId,
+          binding.binding.bindingId,
+          status,
+        );
+
+        return bindingIdentifier + ": failed";
+      }
+    }),
+  );
+}
+
+async function processBinding(
+  instance: ServiceInstance,
+  binding: ServiceBinding,
+  repo: Repository,
+): Promise<string> {
+  const bindingIdentifier = instance.instance.serviceInstanceId + "/" +
+    binding.binding.bindingId;
+
+  console.log("starting to process " + bindingIdentifier);
+
+  try {
+    Deno.statSync(
+      `${repo.path}/terraform/${instance.instance.serviceDefinitionId}`,
+    );
+  } catch (e) {
+    if (e instanceof Deno.errors.NotFound) {
+      console.log(
+        `Skipping ${bindingIdentifier}, because no Terraform folder exists for service definition with id ${instance.instance.serviceInstanceId}`,
+      );
+
+      return bindingIdentifier + ": skipped";
+    }
+  }
+
+  const bindingDir =
+    `${repo.path}/instances/${instance.instance.serviceInstanceId}/bindings/${binding.binding.bindingId}`;
+
+  createTerraformWrapper(instance, binding, bindingDir);
+
+  const tfResult = await executeTerraform(bindingDir);
+
+  updateStatusAfterTfApply(tfResult, repo, instance, binding);
+
+  return tfResult.success
+    ? bindingIdentifier + ": successful"
+    : bindingIdentifier + ": failed";
+}
+
+function createTerraformWrapper(
+  instance: ServiceInstance,
+  binding: ServiceBinding,
+  bindingDir: string,
+) {
+  const terraformWrapper = {
+    variable: {
+      client_secret: {
+        description: "Client Secret of the Azure App Registration.",
+        type: "string",
+        sensitive: true,
+      },
+    },
+    module: {
+      wrapper: {
+        source:
+          `../../../../terraform/${instance.instance.serviceDefinitionId}`,
+        client_secret: "${var.client_secret}",
+        ...instance.instance.parameters,
+        ...binding.binding.bindResource,
+        ...binding.binding.parameters,
```

Comment:
I would be in favor of including the context (customer_id and project_id for meshStack). That makes it easier to name things human-friendly. 

Example: For an Azure Hub-Spoke peering the hub-peering could be called "customer_id-project_id_spoke"

Reply:
makes total sense to provide this information. I added it. Just had to do a little trick to get rid of the auth_urls in the context object as they are not really relevant for terraform SBs imo. They would only be needed if a SB dashboard exists for the SB.